Makefile

```diff
@@ -921,58 +921,57 @@ finish: check-symbols
 endif
 endif
 
-DEFINED_SYMBOLS = $(SYSROOT_SHARE)/defined-symbols.txt
-UNDEFINED_SYMBOLS = $(SYSROOT_SHARE)/undefined-symbols.txt
+install: finish
+	mkdir -p "$(INSTALL_DIR)"
+	cp -r "$(SYSROOT)/lib" "$(SYSROOT)/share" "$(SYSROOT)/include" "$(INSTALL_DIR)"
 
-ifeq ($(WASI_SNAPSHOT),p2)
-EXPECTED_TARGET_DIR = expected/wasm32-wasip2
-else
-ifeq ($(THREAD_MODEL),posix)
-EXPECTED_TARGET_DIR = expected/wasm32-wasip1-threads
-else
-EXPECTED_TARGET_DIR = expected/wasm32-wasip1
-endif
-endif
 
+##### CHECK ####################################################################
+# The `check-symbols` target builds up a set of text files in `<sysroot>/share`
+# which are compared with known-good output in the `expected` directory.
+################################################################################
 
-check-symbols: startup_files libc
-	#
-	# Collect metadata on the sysroot and perform sanity checks.
-	#
+$(SYSROOT_SHARE)/defined-symbols.txt: startup_files libc
 	mkdir -p "$(SYSROOT_SHARE)"
-
-	#
-	# Collect symbol information.
-	#
+	"$(NM)" --defined-only \
+	    $(SYSROOT_LIB)/*.a $(SYSROOT_LIB)/*.o \
+	    |grep ' [[:upper:]] ' \
+	    |sed 's/.* [[:upper:]] //' \
+	    |LC_ALL=C sort \
+	    |uniq \
+	    > "$@"
```

Comment:
Also, I can move all these to separate scripts if we want (here or in a different PR).